packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/TableCellSelection.ts

```diff
@@ -127,4 +134,18 @@ export default class TableCellSelection implements EditorPlugin {
             editor.select(selection.table, null);
         }
     }
+
+    private handleInputEvent(editor: IEditor) {
+        // If typing on any cell selection, clear the first cell
+        const range = editor.getSelectionRangeEx();
+        if (range.type == SelectionRangeTypes.TableSelection) {
```

Comment:
Actually, when I debug your code, Input event happens after KeyDown event. Since you have already changed selection when KeyDown event, it will fail the check of table selection range type, so will go directly to line 149, so code from line 142 to line 146 is never executed.

![image](https://github.com/microsoft/roosterjs/assets/23065085/3a36281f-03a5-4669-b958-b60f5c3f0503)

Reply:
Interesting. I assumed it would retain the selection type.

Reply:
Because you have called editor.select(), that will change selection type.

Reply:
And because of that, the executing result is correct now. But if line 43-46 are executed, the new typed content will be removed, just because when Input event is triggered, the new typed content is already added into the view.

Reply:
I have noticed that, would it work if I did the deletion before the Input event? I would need to add it to PluginEventType.

Reply:
My feeling is you just need to keep you current code to select the content of first selected cell, (and call addUndoSnapshot() in a proper place), then let browser handle everything else, so no need to handle Input event at all.

Because Input event is triggered after type, so anything there is too late.

And maybe only need to handle character types (you can use `isCharacterValue` function to check) so for other keys like caps lock, leave it to browser